recommendation/microsoft.owin.security.json

```diff
@@ -0,0 +1,907 @@
+      "Name": "Microsoft.Owin.Security.IAuthenticationManager.SignOut(Microsoft.Owin.Security.AuthenticationProperties, params string[])",
+      "Value": "Microsoft.Owin.Security.IAuthenticationManager.SignOut(Microsoft.Owin.Security.AuthenticationProperties, params string[])",
+      "KeyType": "",
+      "ContainingType": "IAuthenticationManager",
+      "RecommendedActions": [
+        {
+          "Source": "Amazon",
+          "Preferred": "Yes",
+          "TargetFrameworks": [
+            {
+              "Name": "netcoreapp3.1",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            },
+            {
+              "Name": "net5.0",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            }
+          ],
+          "Description": "For Microsoft.Owin.Security.IAuthenticationManager.SignOut(Microsoft.Owin.Security.AuthenticationProperties, params string[]) add directive Microsoft.AspNetCore.Authentication, add a comment to explain parameters, add an await operator and replace the method SignOut with SignOutAsync.",
+          "Actions": [
+            {
+              "Name": "AddDirective",
+              "Type": "Using",
+              "Value": "Microsoft.AspNetCore.Authentication",
+              "Description": "Add Microsoft.AspNetCore.Authentication namespace",
+              "ActionValidation": {
+                "Contains": "using Microsoft.AspNetCore.Authentication;",
+                "NotContains": ""
+              }
+            },
+            {
+              "Name": "ReplaceMethodAndParameters",
+              "Type": "Method",
+              "Value": {
+                "oldMethod": "SignOut",
+                "newMethod": "SignOutAsync",
+                "newParameters": "()"
+              },
+              "Description": "Replace method name from SignOut to SignOutAsync and remove all parameters.",
+              "ActionValidation": {
+                "Contains": "SignOutAsync",
+                "NotContains": ""
+              }
+            },
+            {
+              "Name": "AddComment",
+              "Type": "Method",
+              "Value": "You can only pass in a single scheme as a parameter and you can also include an AuthenticationProperties object as well.",
+              "Description": "Add a comment to explain how to use the parameters for the SignOutAsync api.",
+              "ActionValidation": {
+                "Contains": "You can only pass in a single scheme as a parameter and you can also include an AuthenticationProperties object as well.",
+                "NotContains": "",
+                "CheckComments": "True"
+              }
+            },
+            {
+              "Name": "AddAwaitOperator",
+              "Type": "Expression",
+              "Value": "",
+              "Description": "Add the await operator to SignOutAsync.",
+              "ActionValidation": {
+                "Contains": "await",
+                "NotContains": ""
```

Comment:
Same as above for `await` -> `await SignOut`

Reply:
See above comment.

Reply:
Thanks Jon, agreed that validation should be tightly coupled with rule as a whole rather than action and any validation logic change would invalidate most validation logic anyway. Made the change.